packages/utils/src/utils/arrays.utils.ts

```diff
@@ -57,5 +57,12 @@ export const hexStringToUint8Array = (hexString: string): Uint8Array => {
   return Uint8Array.from(matches.map((byte) => parseInt(byte, 16)));
 };
 
-export const uint8ArrayToHexString = (bytes: Uint8Array) =>
-  bytes.reduce((str, byte) => str + byte.toString(16).padStart(2, "0"), "");
+export const uint8ArrayToHexString = (bytes: Uint8Array | number[]) => {
+  if (!(bytes instanceof Uint8Array)) {
+    bytes = Uint8Array.from(bytes);
```

Comment:
nice to have: it would be nice to not mutate the provided variables but to realocate for a more functional approach (?).

```
cont b = bytes instanceof Unit8Array ? bytes : Uint8Array.from(bytes);

return b.reduce(...
```

Reply:
It doesn't really make a difference to me but since it's already merged, I'll accept your PR :-).

Reply:
😉